pkg/prometheus/promcfg.go

```diff
@@ -2142,6 +2154,39 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 	}
 }
 
+func generateRoleSelectorConfig(role *monitoringv1alpha1.KubernetesRole, selector *metav1.LabelSelector, k8sSDConfig yaml.MapSlice, cg *ConfigGenerator) yaml.MapSlice {
+	selectors := []yaml.MapSlice{}
+	yml := yaml.MapSlice{
+		{
+			Key:   "role",
+			Value: strings.ToLower(string(*role)),
+		},
+	}
+
+	if len(selector.MatchLabels) > 0 {
+		yml = append(yml, yaml.MapItem{Key: "label", Value: labels.FormatLabels(selector.MatchLabels)})
+		selectors = append(selectors, yml)
+	}
+
+	if len(selector.MatchExpressions) > 0 {
```

Comment:
what if both `MatchLabels` and `MatchExpressions` are defined? should (can?) we merge them?